pkg/controller/acmechallenges/scheduler/scheduler.go

```diff
@@ -200,6 +200,25 @@ func compareChallenges(l, r *cmacme.Challenge) int {
 		return 1
 	}
 
+	// Explicitly allow parallel DNS-01 reqs for the same DNSNames using Route53,
+	// as the route53 package is capable of handling these.
+	// We could assume different solvers are safe to schedule in parallel,
+	// but this might not always be true (e.g., with custom webhooks)
+	if l.Spec.Type == cmacme.ACMEChallengeTypeDNS01 &&
```

Comment:
Hmm.. I really wish we didn't have explicit references to particular DNS-01 provider types in this scheduler package, but can clearly see why it's required unless we can be sure all providers work like this. The only alternative I can see, is we make the scheduler package always allow scheduling multiple challenges for wildcard + non-wildcard and then have each DNS-01 provider perform its own locking/mutex if the provider does not support it.

There's also the case where someone is using a CNAME record for two totally distinct domains pointing to the same name for solving, which is yet another instance where concurrent challenges may be possible assuming provider support.